app/auth/authorize_api_request.rb

```diff
@@ -0,0 +1,28 @@
+class AuthorizeApiRequest
+  def initialize(headers = {})
+    @headers = headers
+  end
+
+  def call
+    { user: user }
+  end
+
+  private 
+
+  attr_reader :headers
+
+  def user
+    @user ||= User.find(decoded_auth_token[:user_id]) if decoded_auth_token
+  rescue ActiveRecord::RecordNotFound => e
+    raise(ExceptionHandler::InvalidToken, ("#{Message.invalid_token} #{e.message}"))
```

Comment:
The parentheses after raise can be left out. Feels more Ruby-y when using `raise`